File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        ans = ""

        for i in range(len(s)):
            ans = max(ans, find(s, i, i), find(s, i, i+1), key=len)
        return ans

def find(s, left, right):
    while left >= 0 and right < len(s) and s[left] == s[right]:
        left -= 1
        right += 1
    return s[left+1:right]
    

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    int[] answer = {0, 1};

    public String longestPalindrome(String s) {
        for (int i = 0; i < s.length(); i++) {
            find(i ,i, s);
            find(i, i + 1, s);
        }
        return s.substring(answer[0], answer[1]);
    }

    public void find(int start, int end, String s) {
        int n = s.length();
        while (start >= 0 && end < n) {
            if (s.charAt(start) == s.charAt(end)) {
                if ((answer[1] - answer[0]) < ((end+1) - start)) {
                    answer[1] = end + 1;
                    answer[0] = start;
                }
                start--;
                end++;
            } else {
                break;
            }
        }
    }
}